src/test/java/io/vertx/test/core/instrumentation/MetricsTest.java

```diff
@@ -205,16 +167,25 @@ public void testHttpServerRequest() throws Exception {
     awaitLatch(latch);
     HttpClientRequest req = client.get(8080, "localhost", "/1", resp -> {
       assertEquals(200, resp.statusCode());
+      List<Span> finishedSpans = tracer.getFinishedSpans();
+      // client request to /1, server request /1, client request /2, server request /2
+      assertEquals(4, finishedSpans.size());
+      assertOneTrace(finishedSpans);
```

Comment:
@vietj now it breaks here.

The metrics impl looks suspicious, when something is stored in thread local (scope created) it should be removed once processing is done - e.g. request is finished.

Reply:
agreed, that's part of the thing we should try to improve